### PR TITLE
[#OP256] Improve nitrogen balance module

### PR DESCRIPTION
**Pull request by elaouniyassine**
### 1. Summary

**Context / Ticket link:**  
Epic #256 on open project

**What does this PR change?**

- Improved nitrogen balance code
- Implemented new nitrogen balance examples
- Improved variable names
- Implemented data checks and ranges
- Implemented nitrogen balance & directemissions manure logic into `run_gleam`

---

### 2. Testing

_The developer is responsible for performing vigilant QA testing before requesting review_

- [x] App / software runs locally without error
- [x] Feature manually tested end-to-end
- [x] The business logic aligns with the corresponding ticket acceptance criteria
- [x] No unintended side effects observed during exploratory testing
- [x] Possible edge cases considered and tested
- [x] Unit tests added or updated (if applicable)

---

### 3. Code Quality Self-Check

_The developer must critically review their own code before requesting review.  
This is a quality gate, not a formality.  
If any box cannot be confidently checked, the PR is not ready._

- [x] Naming and coding style follows [Applitics conventions](https://github.com/APPLITICS/applitics-delivery-system/blob/main/coding_standards.md)
- [x] No merge conflicts, and if conflicts occurred, they have been properly resolved and do not introduce regressions
- [x] No dead code, dead datasets or debug artifacts
- [x] Errors are validated and handled clearly
- [x] No hardcoded secrets or paths
- [x] Structure is modular and readable
- [x] No unnecessary complexity introduced
- [x] Code is self-explanatory and properly documented
- [x] No error or avoidable warning messages are returned in the terminal (warnings must be properly addressed, not suppressed)

---

### 4. Reviewer Focus

_The developer must explicitly indicate where deeper review is needed.  
Highlight areas of uncertainty, complexity, or architectural sensitivity.  
Do not leave this section empty if the PR contains non-trivial logic._

**Please focus specifically on (e.g., logic in X, edge cases around Y, perf in Z):**

> [!IMPORTANT]
> Reviewers must follow the official Applitics Code Review Checklist before approving this PR: [Reviewer Checklist](https://github.com/APPLITICS/applitics-delivery-system/blob/main/qa_code_review_governance.md)

### CODE REVIEW

File: R/run_nitrogen_balance.R [hunk 1]
```diff
@@ -1,76 +1,188 @@
...
-  data[, n_excretion := compute_nitrogen_excretion(
-    animal = Animal_short,
-    n_intake = n_intake,
-    n_retention = n_retention
-  ), by = .I]
+  # --- Step 5: Excretion – N lost (intake - retention) ------------------------
+  cohort_level_data[
+    ,
+    nitrogen_excretion := compute_nitrogen_excretion(
+      species_short = herd_level_data[.SD, on = "herd_id", x.species_short],
```
Comment (lydialanzoni) [inline R/run_nitrogen_balance.R:176]: Maybe this could be simplified if we make animal/species_short required also for the cohort-level dataframe? This would also apply to the other modules.

**Review by lydialanzoni — APPROVED** [review]

**Comment by lydialanzoni** [comment]
No comments on the code, beside the minor one you can find above.
Documentation is also updated and alligned with sharepoint and ready for final revision from @tempiog

**Review by dommens — APPROVED** [review]

**Comment by tempiog** [comment]
@lydialanzoni the first round of documentation revision is done. We need to correct function "compute_nitrogen_retention" for pigs (FA and FS), as discussed:

- [x] inclusion of the new 0.806 correction factor introduced in Equation 10.33A (ipcc 2019) -> dividing birth_weight by 0.806 in lines 214 and 220
- [x] replace parturition_rate with (pregnancy_duration / cohort_duration_days) in line 220
- [x] remove component described in line 219, as FS are not involved in lactation.

**Comment by lydialanzoni** [comment]
> @lydialanzoni the first round of documentation revision is done. We need to correct function "compute_nitrogen_retention" for pigs (FA and FS), as discussed:
> 
> * [x]  inclusion of the new 0.806 correction factor introduced in Equation 10.33A (ipcc 2019) -> dividing birth_weight by 0.806 in lines 214 and 220
> * [x]  replace parturition_rate with (pregnancy_duration / cohort_duration_days) in line 220
> * [x]  remove component described in line 219, as FS are not involved in lactation.

Changes implemented as discussed.
Need to do a final check and to refine the "details" section of the `compute_nitrogen_retention` function for PGS.

**Review by tempiog — APPROVED** [review]

File: R/run_nitrogen_balance.R
Comment (ahmedjoubest) [inline R/run_nitrogen_balance.R:1]: The final object name used in the examples of the `run_*()` functions is inconsistent (`results`, `result`, `module_result`, etc.).
Please standardize this everywhere  (across all modules) in this PR.
The rule is **simple**: functions at the same abstraction level should follow the same naming convention.
Kindly ensure this is aligned consistently across all relevant modules.

**Review by ahmedjoubest — CHANGES_REQUESTED** [review]
Thanks Yassine, we’re almost ready to merge, just one last point.

File: R/run_nitrogen_balance.R
Comment (elaouniyassine) [reply R/run_nitrogen_balance.R:1]: I went with `results` as a name. I improved it in the nitrogen balance and the manure modules here. For energy requirements, it was improved in the enteric PR:
https://github.com/APPLITICS/gleam/pull/122/changes/6977c7624d1b0118467253c03f77f32f05ffcfee#diff-0f97667d7cd4cb0910902bd60fddd1eb791a0bf77f3119270c47fb7fbf42c6bd
